src/common/hex-strings.ts

```diff
@@ -40,8 +42,12 @@ export const isHexString = (inp: any): boolean => {
  * @return Input cast as a buffer.
  */
 export const fromHexString = (inp: Buffer | string): Buffer => {
-  if (typeof inp === 'string' && inp.startsWith('0x')) {
-    return Buffer.from(inp.slice(2), 'hex')
+  if (typeof inp === 'string') {
```

Comment:
Found a bug in this. I think it's useful to assume that strings here will be valid hex strings. Not sure if we should throw when the string is not valid hexadecimal or not.